File: src/main/java/fun/vs/fw/demo/merchantdirectory/MerchantDirectoryConfiguration.java
package fun.vs.fw.demo.merchantdirectory;


import org.springframework.cloud.openfeign.EnableFeignClients;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
@EnableFeignClients
public class MerchantDirectoryConfiguration {

    @Bean
    public MerchantDirectoryAdapter merchantDirectoryAdapter(MerchantDirectoryService merchantDirectoryService) {
        return new MerchantDirectoryAdapter(merchantDirectoryService);
    }
}

File: fvf4k/src/main/kotlin/fvf4k/demo/infra/merchantdirectory/MerchantDirectoryConfiguration.kt
package fvf4k.demo.infra.merchantdirectory

import org.springframework.cloud.openfeign.EnableFeignClients
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration


@Configuration
@EnableFeignClients
class MerchantDirectoryConfiguration {

    @Bean
    fun merchantDirectoryAdapter(merchantDirectoryService: MerchantDirectoryService) =
        MerchantDirectoryAdapter(merchantDirectoryService)
}